lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs

```diff
@@ -102,8 +102,28 @@ impl Client {
     /// It updates the local clipboard cache and returns the encoded message
     /// that should be sent to the RDP server.
     pub fn update_clipboard(&mut self, data: Vec<u8>) -> RdpResult<Vec<u8>> {
+        const CR: u8 = 0x13;
+        const LF: u8 = 0x10;
+
+        // convert LF to CRLF, as required by CF_OEMTEXT
+        let len_orig = data.len();
+        let mut converted = Vec::with_capacity(len_orig);
```

Comment:
Should we allocate more than `len_orig` here in an effort to avoid reallocation? At the very least, we're adding a single null terminator which ensures a reallocation. What we should set it to and whether it is worth it is questionable, but something to consider. [This issue](https://github.com/rust-lang/rust/issues/29931) in the Rust repo describes the challenge. Sounds like Rust currently doubles the length during reallocation.

Reply:
We could. But this is happening on copy, so once per second would be considered fast. I'm not too concerned about it, and we don't really know how many extra bytes we'll need.

Reply:
Right, speed is not an issue. This would only be a problem when `data` is big.